pkg/store/proxy.go

```diff
@@ -147,6 +156,9 @@ func (s *ProxyStore) Series(r *storepb.SeriesRequest, srv storepb.Store_SeriesSe
 		respSender, respRecv, closeFn = newRespCh(gctx, 10)
 	)
 
+	storeCtx, cancel := s.contextWithTimeout(gctx)
```

Comment:
> Also, can I ask what's the point of this logic, what's the use case? E.g WHEN would you set (old) query timeout different to proxy/select/store.timeout underneath PromQL?
> Cannot find any immdiate idea. Do you want to timeout on select, but allow more time for promQL?

Imagine that we have two stores: first is very slow, second is fast. Partial time-out is enabled.
Behaviour that we want to see: if slow store is timed out - we still receiving data from second.
So, to make it possible, store.read-timeout should be less than PromQL timeout.
You're right, we can move it to querier.go and the effect will be the same at the moment (and it was my first implementation :) )

I think it doesn't matter where we have this timeout, more important that at the moment it doesn't work perfect because of reading from stores in mergedSet: reading from stores works in series there and first slow store still blocks reading from second. We need some improvements there...